apps/meteor/app/federation-v2/server/infrastructure/rocket-chat/hooks/index.ts

```diff
@@ -35,7 +35,11 @@ export class FederationHooks {
 	public static canAddFederatedUserToNonFederatedRoom(callback: (user: IUser | string, room: IRoom) => Promise<void>): void {
 		callbacks.add(
 			'federation.beforeAddUserAToRoom',
-			(params: { user: IUser | string }, room: IRoom): void => {
+			(params: { user: IUser | string; inviter: IUser }, room: IRoom): void => {
+				if (!params || !params.user || !params.inviter || !room) {
```

Comment:
```suggestion
			(params: { user: IUser | string; inviter?: IUser }, room: IRoom): void => {
				if (!params || !params.user || !room) {
```
Inviter is optional.